### ISSUE TITLE
Add `allowLineSeparatedGroups` option to `sort-keys` rule

### ISSUE THREAD

**Issue by nicogreenarry** [eslint/eslint#12759]
**What rule do you want to change?**
sort-keys

**Does this change cause the rule to produce more or fewer warnings?**
No changes, if the `allowLineSeparatedGroups` parameter is false or omitted, or fewer warnings if it is true.

**How will the change be implemented? (New option, new default behavior, etc.)?**
Add a new `allowLineSeparatedGroups` option, following the pattern of the [tslint's `object-literal-sort-keys` rule](https://palantir.github.io/tslint/rules/object-literal-sort-keys/).

**Please provide some example code that this change will affect:**
```js
const person = {
  firstName: 'Nico',
  lastName: 'Greenarry',

  hobbies: ['cats', 'dodgeball', 'potlucks'],
};
```

**What does the rule currently do for this code?**
> ESLint: Expected object keys to be in ascending order. 'hobbies' should be before 'lastName'. (sort-keys)

**What will the rule do after it's changed?**
If the `allowLineSeparatedGroups` option is true, then there would be no eslint exception.

**Are you willing to submit a pull request to implement this change?**
Yes. I haven't looked at the eslint codebase before, so I expect there to be a learning curve, but I'm willing to figure out how to implement it properly.

**Comment by nicogreenarry** [comment]
I see that this option is specifically mentioned in the [TSLint Migration Guide](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/ROADMAP.md#maintainability). 
<img width="671" alt="sort-lines in roadmap" src="https://user-images.githubusercontent.com/8515899/71944109-93dfac00-3177-11ea-983f-8c425666cd2d.png">

If this functionality will be completed in due time, then feel free to close this issue. But if it hasn't yet been decided whether to add this option, then consider this my petition to add it!

**Comment by mdjermanovic** [comment]
I support this proposal :+1: 

It hasn't been decided yet. I think this is the same as #12275 which didn't reach [consensus](https://eslint.org/docs/maintainer-guide/issues#consensus) at the time so the issue was closed, but maybe it will be different this time.

This looks to me like a reasonable enhancement, useful to allow for grouping different kinds of properties (which isn't supported by this rule), e.g.:

```js
/*eslint sort-keys: "error"*/

const obj = {
  a: 1,
  b: 2, 
  c: 3,
  
  bar() {}, // error: 'bar' should be before 'c'
  foo() {}
}
```

**Comment by kaicataldo** [comment]
I support this, though I'm not convinced about the `ignore-blank-lines` option name. To me, it currently is ignoring blank lines, and this option would make it account for them. Would an option called `groups` be clear enough? Or `lineSeparatedGroups`?

**Comment by nicogreenarry** [comment]
I see your point about the name. If there isn't a convention (explicit or otherwise) against long-ish option names, I'd prefer something like `lineSeparatedGroups` or `allowLineSeparatedGroups` for clarity.

**Comment by g-plane** [comment]
Do we need a champion for this change?

**Comment by kaicataldo** [comment]
@nicogreenarry In recent times I think we've trended towards longer, more explicit option names :) Do you mind updating the proposal with `allowLineSeparatedGroups`?

@g-plane I'll champion. We just need one more supporter.

**Comment by anikethsaha** [comment]
Marked this as accepted as this has a champion and supporters.

**Comment by JJoGeon** [comment]
Agree with the proposal too. 
@nicogreenarry If you don't mind, can I work on this issue? 
I'm new here and I'm looking for something to contribute. 😄

**Comment by nicogreenarry** [comment]
@JJoGeon yes, by all means!
